src/IotHub/IotHub/IotHub/ManualFailover/InvokeAzureRmIotHubManualFailover.cs

```diff
@@ -0,0 +1,105 @@
+        private const string ResourceIdParameterSet = "ResourceIdSet";
+        private const string ResourceParameterSet = "ResourceSet";
+        private const string InputObjectParameterSet = "InputObjectSet";
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = InputObjectParameterSet,
+            ValueFromPipeline = true,
+            HelpMessage = "Iot Hub Object")]
+        [ValidateNotNullOrEmpty]
+        public PSIotHub InputObject { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ResourceParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Name of the Resource Group")]
+        [ValidateNotNullOrEmpty]
+        [ResourceGroupCompleter]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Iot Hub Resource Id")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = ResourceParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Name of the Iot Hub")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (ParameterSetName.Equals(InputObjectParameterSet))
+            {
+                this.ResourceGroupName = this.InputObject.Resourcegroup;
+                this.Name = this.InputObject.Name;
+            }
+
+            if (ParameterSetName.Equals(ResourceIdParameterSet))
+            {
+                this.ResourceGroupName = IotHubUtils.GetResourceGroupName(this.ResourceId);
+                this.Name = IotHubUtils.GetIotHubName(this.ResourceId);
+            }
+
+            if (ShouldProcess(this.Name, Properties.Resources.ManualFailoverIotHub))
+            {
+                try
+                {
+                    IotHubDescription iotHubDescription = this.IotHubClient.IotHubResource.Get(this.ResourceGroupName, this.Name);
+                    PSIotHub psIotHub = IotHubUtils.ToPSIotHub(iotHubDescription);
+                    string failoverRegion = psIotHub.Properties.Locations.FirstOrDefault(loc => loc.Role.Equals("secondary", StringComparison.OrdinalIgnoreCase)).Location;
+                    FailoverInput failoverInput = new FailoverInput(failoverRegion);
+                    this.IotHubClient.IotHub.ManualFailover(this.Name, failoverInput, this.ResourceGroupName);
+                }
+                catch (Exception e)
```

Comment:
Do not catch and rethrow exce[ptions - this just alters the stack trace, making the exception less useful

Reply:
removed exception block